src/circuit_writer/mod.rs

```diff
@@ -44,6 +45,8 @@ where
     /// 3. During witness generation, the public output computation
     ///    is delayed until the very end.
     pub(crate) public_output: Option<Var<B::Field, B::Var>>,
+
+    ir_writer: ir::IRWriter<B>,
```

Comment:
This is basically only for hints, right? Should we add a comment?

Reply:
Yes, will add more documentations.